MeshLib/MeshSurfaceExtraction.cpp

```diff
@@ -348,4 +327,100 @@ std::vector<MeshLib::Node*> MeshSurfaceExtraction::getSurfaceNodes(
     return sfc_nodes;
 }
 
-} // end namespace MeshLib
+std::vector<MeshLib::Element*> MeshSurfaceExtraction::createSfcElementVector(
+    std::vector<MeshLib::Element*> const& sfc_elements,
+    std::vector<MeshLib::Node*> const& sfc_nodes,
+    std::vector<std::size_t> const& node_id_map)
+{
+    std::vector<MeshLib::Element*> new_elements;
+    new_elements.reserve(sfc_elements.size());
+    for (auto sfc_element : sfc_elements)
+    {
+        unsigned const n_elem_nodes(sfc_element->getNumberOfBaseNodes());
+        auto** new_nodes = new MeshLib::Node*[n_elem_nodes];
+        for (unsigned k(0); k < n_elem_nodes; k++)
+        {
+            new_nodes[k] =
+                sfc_nodes[node_id_map[sfc_element->getNode(k)->getID()]];
+        }
+        if (sfc_element->getGeomType() == MeshElemType::TRIANGLE)
+        {
+            new_elements.push_back(new MeshLib::Tri(new_nodes));
+        }
+        else
+        {
+            assert(sfc_element->getGeomType() == MeshElemType::QUAD);
```

Comment:
make it a strict check with `OGS_FATAL`...

Reply:
:heavy_check_mark: